server/src/main/java/org/elasticsearch/search/fetch/FetchPhase.java

```diff
@@ -172,7 +191,7 @@ public void execute(SearchContext context) {
             }
 
             for (FetchSubPhase fetchSubPhase : fetchSubPhases) {
-                fetchSubPhase.hitsExecute(context, hits);
+                fetchSubPhase.hitsExecute(context, sortedHits);
```

Comment:
It would be nice if these fetch sub phases, which implement `hitsExecute` (plural), could also benefit from locality. Currently they each loop through the hits array separately, so any cached data from processing a hit may be lost by the time the next fetch phase is run.

To me this should be a separate change, I can file another issue about it.